constantine/serialization/codecs_banderwagon.nim

```diff
@@ -44,6 +46,21 @@ func validate_scalar*(scalar: matchingOrderBigInt(Banderwagon)): CttCodecScalarS
     return cttCodecScalar_ScalarLargerThanCurveOrder
   return cttCodecScalar_Success
 
+func make_scalar_mod_order*(reduced_scalar: var Fr[Banderwagon], src: array[32, byte], order: static Endianness = bigEndian): bool =
+  var res: bool = false
+  var scalar {.noInit.}: BigInt[256]
+  scalar.unmarshal(src, order)
+
+  getMont(reduced_scalar.mres.limbs, scalar.limbs,
+        Fr[Banderwagon].fieldMod().limbs,
+        Fr[Banderwagon].getR2modP().limbs,
+        Fr[Banderwagon].getNegInvModWord(),
+        Fr[Banderwagon].getSpareBits())
+  res = true
+  return res
```

Comment:
Was this supposed to be used by `fromVerkleDigest`?

Reply:
i have kept this inorder to have an upstreamed general-purpose function, if while writing the verkle trie we need to `deserialize_scalar_mod_order`, yes it does have some code reuse, but i thought it can be a well-named abstraction.

Reply:
I'm keeping one of them and removing fromVerkleDigest in that case

Reply:
It's fine to keep both and have fromVerkleDigest call the other one.

Reply:
The generalization part is something I thought about as well.

From the top of my head it's at least used for EIP2333 implementation (BLS Key Derivation / Lamport signatures) and I think it is used in hashing-to-curve for hashing to field.

Reply:
Cool,then it stays